packages/sdk/src/channel-manager/channel.ts

```diff
@@ -604,3 +622,215 @@ export async function setBaseURI(
+  channelId,
+  commentData,
+  metadata,
+  msgSender,
+  readContract,
+  channelManagerAddress,
+}: GetEstimatedChannelPostCommentHookFeeParams): Promise<HookFeeEstimation> {
+  const channelInfo = await getChannel({
+    channelId,
+    channelManagerAddress,
+    readContract: readContract,
+  });
+
+  if (!channelInfo.hook) {
+    return {
+      amount: 0n,
+      asset: NATIVE_ASSET_ADDRESS,
+      description: "",
+      metadata: [],
+    };
+  }
+
+  return await readContract({
+    abi: BaseHookABI,
+    address: channelInfo.hook,
+    functionName: "estimateAddCommentFee",
+    args: [commentData, metadata, msgSender],
+  });
+}
+
+type GetEstimatedChannelEditCommentHookFeeParams = {
+  channelId: bigint;
+  commentData: CommentData;
+  metadata: MetadataEntry[];
+  msgSender: Hex;
+  readContract: HookContractReadFunctions["estimateEditCommentFee"] &
+    ContractReadFunctions["getChannel"];
+  channelManagerAddress?: Hex;
+};
+
+/**
+ * It calls the `estimateEditCommentFee` function on the hook to retrieve the estimated fee for editing a comment to a channel.
+ * For estimation of total fee, use the `estimateChannelEditCommentFee` helper
+ *
+ * @param getEstimateChannelEditCommentHookFeeParams - The parameters for estimating the fee for editing a comment to a channel
+ * @returns The estimated fee for editing a comment to a channel
+ */
+export async function getEstimatedChannelEditCommentHookFee({
+  channelId,
+  commentData,
+  metadata,
+  msgSender,
+  readContract,
+  channelManagerAddress,
+}: GetEstimatedChannelEditCommentHookFeeParams): Promise<HookFeeEstimation> {
+  const channelInfo = await getChannel({
+    channelId,
+    channelManagerAddress,
+    readContract: readContract,
+  });
+
+  if (!channelInfo.hook) {
+    return {
+      amount: 0n,
+      asset: NATIVE_ASSET_ADDRESS,
+      description: "",
+      metadata: [],
+    };
+  }
+
+  return await readContract({
+    abi: BaseHookABI,
+    address: channelInfo.hook,
+    functionName: "estimateEditCommentFee",
+    args: [commentData, metadata, msgSender],
+  });
+}
+
+type EstimateChannelCommentActionFeeParams<
+  CommentActionFunc extends
+    | typeof getEstimatedChannelPostCommentHookFee
+    | typeof getEstimatedChannelEditCommentHookFee,
+> = {
+  readContract: ContractReadFunctions["getHookTransactionFee"] &
+    ERC165ContractReadFunctions["supportsInterface"] &
+    ERC20ContractReadFunctions["name"] &
+    ERC20ContractReadFunctions["symbol"] &
+    ERC20ContractReadFunctions["decimals"] &
+    ERC20ContractReadFunctions["totalSupply"];
+  commentActionFunc: CommentActionFunc;
+} & (CommentActionFunc extends typeof getEstimatedChannelPostCommentHookFee
+  ? GetEstimatedChannelPostCommentHookFeeParams
+  : GetEstimatedChannelEditCommentHookFeeParams);
+
+/**
+ * Base function for best-effort estimation of the total fee for a comment action to a channel
+ * @param param0
+ * @returns
+ */
+async function estimateChannelCommentActionFee<
```

Comment:
hi @davidfurlong , this function helps to pull the total fee for posting/editing on a channel with hook.
do we want to add code to cover existing channels that have not yet implement `IFeeEstimatble`? such as current Takes channel?